src/tools/solver_manager.h

```diff
@@ -0,0 +1,34 @@
+#ifndef EPIC_SOLVER_MANAGER_H
+#define EPIC_SOLVER_MANAGER_H
+
+#include "concurrent_container.h"
+#include "solver_protocol.h"
+#include "solver_task.h"
+#include "threadpool.h"
+
+
+class SolverManager {
+public:
+    explicit SolverManager(size_t nThreads);
+    TaskStatus ProcessTask(const std::shared_ptr<SolverTask>& task);
+    TaskStatus Solve(std::shared_ptr<SolverTask> task);
+    void AbortTask(uint32_t task_id);
+
+    bool Start();
+    bool Stop();
+
+private:
+    std::atomic_bool enabled_ = false;
+    BlockingQueue<uint32_t> task_queue_;
```

Comment:
I think the type of `task_queue_` can be std::queue

Reply:
I think `BlockingQueue` here is necessary because the entry of the `task_queue_ ` may be accessed by different worker threads of grpc server